### PR TITLE
add a changelog link to the updater

### PR DESCRIPTION
**Pull request by violoncelloCH**
fixes nextcloud/server#6117
this generates a link to the changelog like nextcloud.com/changelog/#X-X-X and adds it to the updater

Signed-off-by: Jonas Sulzer <jonas@violoncello.ch>

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -533,6 +535,18 @@ public function createBackup() {
...
 	}
 
+	/**
+	* Returns a link to the changelog
+	* @return string
+	*
+	*/
+	private function getChangelogURL() {
+		$this->silentLog('[info] getChangelogURL()');
+		$changelogWebsite = "https://nextcloud.com/changelog/";
```
Comment (danielkesselberg) [inline index.php]: Please use Single Quotes here

File: index.php [hunk 2]
```diff
@@ -533,6 +535,18 @@ public function createBackup() {
...
+		$changelogURL = $changelogWebsite . '#' . str_replace('.','-', substr($this->getConfigOption('version'),0,-2));
```
Comment (danielkesselberg) [inline index.php]: I would add a single whitespace after the comma. Just my personal preference but easier to read.

`str_replace('.', '-', substr($this->getConfigOption('version'), 0, -2)` 

Most examples from https://docs.nextcloud.com/server/14/developer_manual/general/codingguidelines.html looks like that.

**Comment by MorrisJobke** [comment]
Could you rebase your branch on latest master to get the fixed tests from #196. ;) If you need help I'm happy to help you.

File: index.php [hunk 3]
```diff
@@ -241,7 +243,7 @@ public function checkForUpdate() {
 		if ($version !== '' && $version !== $this->currentVersion) {
 			$this->updateAvailable = true;
 			$releaseChannel = $this->getCurrentReleaseChannel();
-			$updateText = 'Update to ' . htmlentities($versionString) . ' available. (channel: "' . htmlentities($releaseChannel) . '")<br /><span class="light">Following file will be downloaded automatically:</span> <code class="light">' . $response['url'] . '</code>';
+			$updateText = 'Update to ' . htmlentities($versionString) . ' available. (channel: "' . htmlentities($releaseChannel) . '")<br /><span class="light">Following file will be downloaded automatically:</span> <code class="light">' . $response['url'] . '</code><br /><a class="external_link" href="' . $changelogURL . '" target="_blank" rel="noreferrer noopener">Open changelog ↗</a>';
```
Comment (nickvergessen) [inline index.php]: I would recommend to not use an emoji until we ship a font

File: index.php (hunk 1 above)
Comment (violoncelloch) [reply index.php]: changed -- thank you!

File: index.php (hunk 2 above)
Comment (violoncelloch) [reply index.php]: changed -- thank you!

**Comment by violoncelloch** [comment]
> Could you rebase your branch on latest master to get the fixed tests from #196. ;) If you need help I'm happy to help you.

done -- thank you for your help @MorrisJobke

File: index.php [hunk 4]
```diff
@@ -533,6 +535,18 @@ public function createBackup() {
...
+	* @return string
+	*
+	*/
+	private function getChangelogURL() {
+		$this->silentLog('[info] getChangelogURL()');
+		$changelogWebsite = 'https://nextcloud.com/changelog/';
+		$changelogURL = $changelogWebsite . '#' . str_replace('.', '-', substr($this->getConfigOption('version'),0,-2));
+		return $changelogURL;
+	}
+
```
Comment (MorrisJobke) [inline index.php]: Could you add all those changes to the class also here to the file `Updater.php` to have parity in the CLI and web UI.

Comment (violoncelloch) [reply index.php]: Done!

**Comment by MorrisJobke** [comment]
I updated the PR to show the URL for the upcoming (and not the installed) release and to show it only for the stable releases (we don't have changelogs for betas and RCs)

**Comment by violoncelloCH** [comment]
@MorrisJobke did you intend to close the backport PRs?

**Comment by MorrisJobke** [comment]
> @MorrisJobke did you intend to close the backport PRs?

Yes - thought about it and don't want to risk breakage due to this little link. It's in 16+ and on the pre-updater page for older versions.

**Comment by violoncelloCH** [comment]
makes sense... thank you for explaining!

**Pull request merged**